examples/service-bus/main.tf

```diff
@@ -0,0 +1,218 @@
+resource "humanitec_application" "example" {
+  id   = var.name
+  name = var.name
+}
+
+locals {
+  # Classes used to build the resource definition graph
+  service_bus_basic_class            = "basic"
+  service_bus_publisher_policy_class = "service-bus-basic-publisher"
+  service_bus_consumer_policy_class  = "service-bus-basic-consumer"
+
+  # Classes that developers can select from
+  service_bus_publisher_class = "basic-publisher"
+  service_bus_consumer_class  = "basic-consumer"
+
+  scope = "/subscriptions/${var.subscription_id}/resourceGroups/${var.resource_group_name}/providers/Microsoft.ServiceBus/namespaces/$${resources['azure-service-bus-queue.${local.service_bus_basic_class}'].outputs.namespace}/queues/$${resources['azure-service-bus-queue.${local.service_bus_basic_class}'].outputs.queue}"
+}
+
+module "service_bus" {
+  source = "../../humanitec-resource-defs/service_bus/basic"
+
+  resource_packs_azure_url = var.resource_packs_azure_url
+  resource_packs_azure_rev = var.resource_packs_azure_rev
+  client_id                = var.client_id
+  client_secret            = var.client_secret
+  tenant_id                = var.tenant_id
+  subscription_id          = var.subscription_id
+  resource_group_name      = var.resource_group_name
+  prefix                   = var.prefix
+  sku                      = var.sku
+}
+
+resource "humanitec_resource_definition_criteria" "service_bus" {
+  resource_definition_id = module.service_bus.id
+  app_id                 = humanitec_application.example.id
+  class                  = local.service_bus_basic_class
+}
+
+// Publisher shared
+
+module "service_bus_publisher" {
+  source = "../../humanitec-resource-defs/service_bus/passthrough"
+
+  prefix                     = "${var.prefix}publisher-"
+  policy_resource_class      = local.service_bus_publisher_policy_class
+  service_bus_resource_class = local.service_bus_basic_class
+}
+
+resource "humanitec_resource_definition_criteria" "service_bus_publisher" {
+  resource_definition_id = module.service_bus_publisher.id
+  app_id                 = humanitec_application.example.id
+  class                  = local.service_bus_publisher_class
+}
+
+// Copy of Azure Service Bus Data Sender
+module "role_definition_publisher" {
+  source = "../../humanitec-resource-defs/role_definition/basic"
+
+  prefix = "${var.prefix}publisher-"
+
+  resource_packs_azure_url = var.resource_packs_azure_url
+  resource_packs_azure_rev = var.resource_packs_azure_rev
+  client_id                = var.client_id
+  client_secret            = var.client_secret
+  tenant_id                = var.tenant_id
+  subscription_id          = var.subscription_id
+  scope                    = local.scope
+  actions = [
+    "Microsoft.ServiceBus/*/queues/read",
+    "Microsoft.ServiceBus/*/topics/read",
+    "Microsoft.ServiceBus/*/topics/subscriptions/read"
+  ]
+  data_actions = [
+    "Microsoft.ServiceBus/*/send/action"
+  ]
```

Comment:
Not sure if Azure has any recommendations here, but I wonder if using the existing role wouldn't be recommended over creating on manually https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#azure-service-bus-data-sender 🤔